src/history.rs

```diff
@@ -72,6 +72,7 @@ const_assert_eq!(
 /// backed by runtime-sized arrays of current logical clocks
 /// and probe log items
 #[derive(Debug)]
+#[repr(C)]
```

Comment:
This is a step in the right direction for sorting out a fixed layout. `FixedSliceVec` will need to be updated to be `repr(C)` for this to be complete due to its use in the `clocks` and `compact_log` fields. All of the other fields are `repr(C)` or have transparent primitive layouts.

Reply:
Yes, `FixedSliceVec` will need to become `repr(C)`, but I was holding off on dealing with that until I update DynamicHistory, which is not in the scope of this PR.